modules/browsemap/src/main/java/com/sygic/modules/browsemap/viewmodel/BrowseMapFragmentViewModel.kt

```diff
@@ -1,105 +1,115 @@
+    @MapSelectionMode
+    var mapSelectionMode: Int = MapSelectionMode.MARKERS_ONLY
     val compassEnabled: MutableLiveData<Boolean> = MutableLiveData()
     val compassHideIfNorthUp: MutableLiveData<Boolean> = MutableLiveData()
     val positionLockFabEnabled: MutableLiveData<Boolean> = MutableLiveData()
     val zoomControlsEnabled: MutableLiveData<Boolean> = MutableLiveData()
 
     val poiDataObservable: LiveData<PoiData> = SingleLiveEvent()
 
+    val dialogFragmentListener: DialogFragmentListener = object : DialogFragmentListener {
+        override fun onDismiss() {
+            extendedMapDataModel.removeOnClickMapMarker()
+        }
+    }
+
+    private var onMapClickListener: OnMapClickListener? = null
+
     init {
         attributesTypedArray?.let {
+            mapSelectionMode =
+                    it.getInt(R.styleable.BrowseMapFragment_sygic_map_selectionMode, MapSelectionMode.MARKERS_ONLY)
             compassEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compass_enabled, false)
             compassHideIfNorthUp.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compass_hideIfNorthUp, false)
-            mapInteractionMode.value =
-                    it.getInt(R.styleable.BrowseMapFragment_sygic_map_interactionMode, MapInteractionMode.MARKERS_ONLY)
             positionLockFabEnabled.value =
                     it.getBoolean(R.styleable.BrowseMapFragment_sygic_positionLockFab_enabled, false)
             zoomControlsEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_zoomControls_enabled, false)
             it.recycle()
         }
 
-        mapInteractionMode.value?.let { mode ->
-            when (mode) {
-                MapInteractionMode.FULL, MapInteractionMode.MARKERS_ONLY -> mapInteractionManager.addOnMapClickListener(
-                    this
-                )
-            }
-        }
+        mapInteractionManager.addOnMapClickListener(this)
     }
 
     override fun onMapObjectsRequestStarted() {
         extendedMapDataModel.removeOnClickMapMarker()
     }
 
     override fun onMapObjectsReceived(viewObjects: List<ViewObject>) {
-        mapInteractionMode.value?.let { mode ->
-            val firstViewObject = viewObjects.first()
-            when (mode) {
-                MapInteractionMode.NONE -> return
-                MapInteractionMode.MARKERS_ONLY -> {
-                    if (firstViewObject is MapMarker) {
-                        getPoiData(firstViewObject)
-                        return
-                    }
-
-                    notifyPoiDataChanged(PoiData.EMPTY)
+        when (mapSelectionMode) {
+            MapSelectionMode.NONE -> {
+                logWarning("NONE")
+            }
+            MapSelectionMode.MARKERS_ONLY -> {
+                val firstViewObject = viewObjects.first()
+                if (firstViewObject !is MapMarker) {
+                    logWarning("MARKERS_ONLY")
```

Comment:
toto nevidim ako nutny warning. ano pri NONE, ked nastavi listener a ma none, tak je to zvlastna kombinacia, ale ak nastavi pri MARKERS_ONLY a netrafi marker, tak to zas nie je nieco co sa nemoze bezne stat...

Reply:
Chápu, jde jen o to, že MARKERS_ONLY je by default, pokud nic nenastaví. Myslel jsem, aby se nedivil, že si nastavil onMapClickListener a chodí mu jen kliky na jeho markery ;)